sgxs-tools/src/sgx_detect/imp/windows.rs

```diff
@@ -0,0 +1,40 @@
+use failure::Error;
+
+use crate::DetectError;
+use crate::interpret::{AesmStatus, KmodStatus};
+use winapi::um::winbase::GetFirmwareEnvironmentVariableA;
+use winapi::shared::minwindef::DWORD;
+use std::ffi::CString;
+
+pub fn rdmsr(_address: u64) -> Result<u64, Error> {
+    bail!("RDMSR not implemented on Windows")
+}
+
+pub fn read_efi_var(name: &str, guid: &str) -> Result<Vec<u8>, Error> {
+    let mut env: Vec<u8> = vec![0; 1024];
+    let name = CString::new(name)?.as_ptr();
```

Comment:
You should store the `CString` in the variable and call `as_ptr` below when calling the C function. The way you've written it now, the `CString` is deallocated at the end of the statement.